app/models/concerns/digital_object_concerns/save_behavior.rb

```diff
@@ -52,6 +52,7 @@ def save_impl(opts = {})
         begin
           self.mint_uid_and_metadata_location_uri_if_new_record
           self.update_modification_info(current_datetime, opts[:user])
+          self.mint_reserved_doi_if_doi_blank
```

Comment:
If something goes wrong during the save, the digital object's doi field will be reverted properly because of the existing code below, but I think the DOI will still exist indefinitely in a reserved state in the external DOI system.

Reply:
I'm not sure what to do about this. Send the UID with the DOI mint metadata? Reluctant to rely on sending a withdrawal since we'd already be in an error state. We could have a pending DOI and do the mint after the save?

Reply:
I'm opening a separate issue for this question.

Reply:
Sounds good.  As long as we have an issue, I'm fine with not solving with this PR.